orttraining/orttraining/test/training_api/core/training_api_tests.cc

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <thread>
+#include <random>
+
+#include "gtest/gtest.h"
+#include "nlohmann/json.hpp"
+
+#include "test/framework/test_utils.h"
+#include "test/util/include/asserts.h"
+#include "core/common/path_utils.h"
+#include "core/framework/tensorprotoutils.h"
+#include "orttraining/training_api/include/utils.h"
+#include "orttraining/training_api/include/module.h"
+#include "orttraining/training_api/include/optimizer.h"
+#include "orttraining/training_api/include/checkpoint_property.h"
+#include "orttraining/training_api/include/checkpoint.h"
+#include "orttraining/training_api/include/lr_scheduler.h"
+#include "orttraining/test/training_api/core/data_utils.h"
+#include "default_providers.h"
+
+using json = nlohmann::json;
+using namespace onnxruntime::training;
```

Comment:
Have we considered the lints on using? They tend to pollute namespaces? https://google.github.io/styleguide/cppguide.html#Namespaces

Reply:
Although this should be addressed, I will get it in a future update to avoid delaying this merge anymore.

Reply:
Addressed this.